rfc/0009-audit-checks.md

```diff
@@ -0,0 +1,218 @@
+additional functions. However, the policies still need to be updated to verify if
+this field is populated and, if it is, verify the resources there.
+
+Furthermore, the `policy-evaluator` should also be updated to propagate this field
+until the Policy Server where the audit checks will be trigger and the results will be processed.
+For that, the types used to store the policies responses as well as the functions used
+to process these types and to send the validation request to the policies, must be updated
+to add the new fields used to store the audit checks resources and results.
+
+## Audit check results
+
+Once the policy server has the audit checks results, it will populate the a [PolicyReport](https://github.com/kubernetes-sigs/wg-policy-prototypes/tree/master/policy-report).
+The report will contains all the success and failures  returned by the policies.
+Results from `ClusterAdmissionPolicy` (cluster wide policies) will be stored in the `ClusterPolicyReport`
+and the result from the `AdmissionPolicy` (namespaces policies) will be stored in the `PolicyReport`.
+
+The reports should contains some minimum information listed below:
+
+ - Policy name/ID
+ - Policy server name/ID
+ - Policy validation result (e.g. `success`, `fail`, etc)
+ - Policy failure error message, if it exists. The same error message shown to the users if they try to apply an invalid resource in the cluster.
+ - Resource identification
+ - Policy mode ( `monitor` or `protect` mode)
+ - Timestamp of the validation
+
+Any additional information is optional.
+
+# Drawbacks
+
+It's not possible to define if a policy with `audit` field `true` is really
+validating the resources in the new added `resources` field before running the audit check.
+
+Download policies one more time. Once for the policy server running in audit mode
+and others for the other policy servers
+
+# Alternatives
+
+Additional to the `PolicyReport` report, policy server could add an annotation to the
+resources violating policies. However, this will require additional permissions to the policy server.
+
+Policies could have a dedicated function to evaluate resources in the audit checks
+
+If the number of resources for an individual policy is too high, the resources could
+be batched and multiple instances of the same policy could be run at the same time
+temporally to speed up the policy validations.  This horizontal scaling for all the
+policies under demand is a good topic for another RFC or idea discussion for the future.
+
+
+Run the audit checks in the same policy server where the admissions requests are
+evaluated. And use cronjob to trigger audit checks. But considering [that cronjob
+can sometimes trigger none or twice a job](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-job-limitations),
+plus some benefits like avoiding competing resources, a separated policy server seems a better idea
+
```

Comment:
Another alternative similar to this one:
Run the audit policy server in a cron job. Then we don't overload existing PolicyServers. It's rare that cronjobs are triggered none or twice, but if that happens it wouldn't be such a big deal. If it not triggered it would be trigger next time, and if it is triggered twice 2 reports would be created. I don't think this will happen very often (if it happens at all)
I don't like the idea of having one policy server waiting for audit reports that might happen once a day. I think CronJobs are a good use case for generating [reports](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cronjob)
Another good point of this approach is that a user could manually create a job and generate a report at any time easily.

Reply:
Okay. In my first thoughts I believed that a missing audit checks could be a big deal. But if it is not, and the majority of the team agree up on this, I'm fine with using cronjobs. Actually, while I was writing this RFC this was one of my first options. But I've changed mind later